notebook_demo.ipynb

```diff
@@ -395,19 +395,24 @@
    "source": [
     "from prompt2model.dataset_processor import TextualizeProcessor\n",
     "\n",
-    "t5_processor = TextualizeProcessor(has_encoder=True)\n",
-    "t5_modified_dataset_dicts = t5_processor.process_dataset_lists(\n",
-    "    prompt_spec.instruction, [generated_dataset, retrieved_dataset_dict[\"train\"]], train_proportion=0.6, val_proportion=0.2, maximum_exmaple_num=3000\n",
-    ")"
+    "text_processor = TextualizeProcessor(has_encoder=True)\n",
+    "text_modified_dataset_dicts = text_processor.process_dataset_lists(\n",
+    "    prompt_spec.instruction,\n",
+    "    [generated_dataset, retrieved_dataset_dict[\"train\"]],\n",
+    "    train_proportion=0.6,\n",
+    "    val_proportion=0.2,\n",
+    "    maximum_exmaple_num=3000\n",
+    ")\n",
+    "train_datasets = [each[\"train\"] for each in text_modified_dataset_dicts]\n",
+    "val_datasets = [each[\"val\"] for each in text_modified_dataset_dicts]\n",
+    "test_datasets = [each[\"test\"] for each in text_modified_dataset_dicts]\n"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Finetune the Model\n",
-    "\n",
-    "Combine the retrieved dataset with generated dataset and use the `GenerationModelTrainer` to finetune the retrieved model. After the finetuning, we save the model and tokenizer to the disk."
+    "Then we use the `GenerationModelTrainer` to finetune the retrieved model and save the model to disk. If you're running on Colab you can then download the model to your local machine."
```

Comment:
BTW, if we merge this line into `add_colab_demo`, this sentence suggesting "download to local machine" is somehow redundant. 🤔